### PR TITLE
Redis cache

### PR DESCRIPTION
**Pull request by MihailNenahov**
Added Serializable to each class
Added Cache

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,14 +2,13 @@ version: "3.8"
 services:
   mongo:
     image: mongo:4.4
-    container_name: mongo
+    container_name: mongo_l
```
Comment (arramanenka) [inline docker-compose.yml:5]: ?

File: docker-compose.yml [hunk 2]
```diff
@@ -2,14 +2,13 @@ version: "3.8"
...
     hostname: mongo
     ports:
       - "27017:27017"
   redis:
     image: "redis:alpine"
+    container_name: redis_l
```
Comment (arramanenka) [inline docker-compose.yml:11]: why _1? why do you even specify name for container?

File: src/main/java/com/logistic/project/Application.java [hunk 3]
```diff
@@ -2,7 +2,9 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cache.annotation.EnableCaching;
 
+@EnableCaching
```
Comment (arramanenka) [inline src/main/java/com/logistic/project/Application.java:7]: are you sure this is needed?

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -3,6 +3,12 @@ spring.data.mongodb.host=localhost
 spring.data.mongodb.port=27017
 spring.data.mongodb.database=logistic_database
 
+#redis
+spring.cache.type=redis
+spring.redis.host=localhost
```
Comment (arramanenka) [inline src/main/resources/application.properties]: default parameters can be ommitted

File: src/main/java/com/logistic/project/controller/LogisticController.java [hunk 5]
```diff
@@ -25,18 +27,21 @@
     private final OrderRepository orderRepository;
 
     @GetMapping("/orders")
+    @Cacheable(value = "List<Order>", key = "{#customerName, #page, #size}")
```
Comment (arramanenka) [inline src/main/java/com/logistic/project/controller/LogisticController.java]: What if someone adds an order after hitting this endpoint once?

File: docker-compose.yml (hunk 1 above)
Comment (MihailNenahov) [reply docker-compose.yml:5]: just to show that this container created for logistic servcie

File: src/main/java/com/logistic/project/Application.java (hunk 3 above)
Comment (MihailNenahov) [reply src/main/java/com/logistic/project/Application.java:7]: AutoConfiguration. Or it's not required?

**Review by arramanenka — APPROVED** [review]

**Pull request merged**